test/tests/functional/pbs_resource_usage_log.py

```diff
@@ -0,0 +1,350 @@
+        matches = self.server.accounting_match(
+            msg='.*R;' + jid + '.*Exit_status=0.*resources_used.*run_count=1',
+            id=jid, regexp=True, allmatch=True)
+        self.assertEqual(len(matches), 2, " 2 R records expected")
+        self.server.accounting_match(
+            msg='.*E;' + jid +
+            '.*Exit_status=271.*resources_used.*run_count=2',
+            id=jid, regexp=True)
+
+        # job array's subjobs have 2 R records and
+        # the jobarray has E record with run_count=0
+        matches = self.server.accounting_match(
+            msg='.*R;' + re.escape(subjid1) +
+            '.*Exit_status=0.*resources_used.*run_count=1',
+            id=subjid1, regexp=True, allmatch=True)
+        self.assertEqual(len(matches), 2, " 2 R records expected")
+        matches = self.server.accounting_match(
+            msg='.*R;' + re.escape(subjid2) +
+            '.*Exit_status=0.*resources_used.*run_count=1',
+            id=subjid2, regexp=True, allmatch=True)
+        self.assertEqual(len(matches), 2, " 2 R records expected")
+        self.server.accounting_match(
+            msg='.*E;' + re.escape(jid_a) +
+            '.*Exit_status=1.*run_count=0', id=jid_a,
+            regexp=True)
+
+        # Set to defaults
+        a = {ATTR_nodefailrq: 310}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+    def test_acclog_job_multiple_qrerun(self):
+        """
+        Check for R record in accounting logs when job is
+        requeued using qrerun command
+        """
+        a = {'resources_available.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+
+        # Submit job
+        a = {'Resource_List.select': '1:ncpus=1:mem=20gb'}
+        j = Job(TEST_USER, a)
+        j.create_eatcpu_job()
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid)
+
+        # Submit job array
+        ja = Job(TEST_USER, attrs={
+            ATTR_J: '1-2',
+            'Resource_List.select': 'ncpus=1:mem=20gb'}
+        )
+        ja.create_eatcpu_job()
+        jid_a = self.server.submit(ja)
+        subjid1 = j.create_subjob_id(jid_a, 1)
+        subjid2 = j.create_subjob_id(jid_a, 2)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid1)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid2)
+
+        # Rerun jobs first time
+        self.server.rerunjob(jobid=jid)
+        self.server.rerunjob(jobid=jid_a)
+
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid1)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid2)
+        time.sleep(30)
+
+        # Rerun jobs second time
+        self.server.rerunjob(jobid=jid)
+        self.server.rerunjob(jobid=jid_a)
+        self.server.expect(JOB, {ATTR_substate: '42'}, jid)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid1)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid2)
```

Comment:
You are making an assumption here that the jobs were requeued and the scheduler reran them.  You should turn off scheduling before you requeue the jobs so you can catch them in the Q state.  Once you do, turn scheduling on to see them back in the R state.  Otherwise, you don't know if the requeue failed and they are still in the R state from the original run.

Reply:
Done